System/Writer.h

```diff
@@ -1,322 +1,173 @@
-        for (int i = 0; i < m_Indent; ++i)
-            *m_pStream << "\t";
-    }
-
-/*
-//////////////////////////////////////////////////////////////////////////////////////////
-// Manipulator:     newl
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Creates a new line that is properly indented.
-// Arguments:       An ostream reference.
-// Return value:    An ostream reference.
-
-    static std::ostream & newl(std::ostream &stream)
-    {
-        stream << "\n";
-        for (int i = 0; i < m_Indent; ++i)
-            stream << "\t";
-        return stream;
-    }
-*/
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Protected member variable and method declarations
-
-protected:
-
-    // Member variables
-    static const std::string ClassName;
-    std::ofstream *m_pStream;
-    // Currently used stream's filepath
-    std::string m_FilePath;
-    // Only the name of the currently read file, excluding the path
-    std::string m_FileName;
-    // Only the path to the folder that we are writing a file in, excluding the filename
-    std::string m_FolderPath;
-//    zip::ozipfile *m_Package;
-    int m_Indent;
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Private member variable and method declarations
-
-private:
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Clear
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Clears all the member variables of this Writer, effectively
-//                  resetting the members of this abstraction level only.
-// Arguments:       None.
-// Return value:    None.
-
-    void Clear();
-
-
-    // Disallow the use of some implicit methods.
-    Writer(const Writer &reference);
-    Writer & operator=(const Writer &rhs);
-
-};
-
-} // namespace RTE
-
-#endif // File
\ No newline at end of file
+namespace RTE {
+
+	const char g_WritePackageExtension[8] = ".rte";
+
+	/// <summary>
+	/// Writes RTE objects to std::ostreams.
+	/// </summary>
+	class Writer {
+
+	public:
+
+#pragma region Creation
+		/// <summary>
+		/// Constructor method used to instantiate a Writer object in system memory. Create() should be called before using the object.
+		/// </summary>
+		Writer() { Clear(); }
+
+		/// <summary>
+		/// Constructor method used to instantiate a Writer object in system memory. Create() should be called before using the object.
+		/// </summary>
+		/// <param name="filename">Path to the file to open for writing.</param>
+		/// <param name="append">Whether to append to the file if it exists, or to overwrite it.</param>
+		Writer(const char *filename, bool append = false) { Clear(); Create(filename, append); }
+		Writer(std::string filename, bool append = false) { Clear(); Create(filename.c_str(), append); }
```

Comment:
You can't call `c_str()` on a `const char*`.

Reply:
Yeah, other way around. If I wanted to make a writer and I have a std::string filename instead of a const char*, I'd do what's being done here and convert it.
```
std::string myFilename = "hithere.ini";
Writer w = Writer(myFilename);
```
Becomes
```
std::string myFilename = "hithere.ini";
Writer w = Writer(myFilename.c_str());
```

Reply:
I don't know how much changes this might drag in other files.
So i'll let you handle this one if you feel like it.